### PR TITLE
(SERVER-2836) Add release notes for 6.12.1

### PR DESCRIPTION
**Pull request by mwaggett**

### CODE REVIEW

**Comment by mwaggett** [comment]
Would love to simplify these notes.

File: documentation/release_notes.markdown [hunk 1]
```diff
@@ -10,6 +10,16 @@ canonical: "/puppetserver/latest/release_notes.html"
 [puppetserver.conf]: ./config_file_puppetserver.markdown
 [product.conf]: ./config_file_product.markdown
 
+## Puppet Server 6.12.1
+
+Released 14 July 2020
+
+### Enhancements
+
+- Users who’ve started logging jolokia debug info will now have to enable a setting in the config file to get that back. New users that want to get debug info will have to add that setting, in additional updating logback.xml as they would have had to do before. Now, by default, attempting to access uncreated metrics will quietly continue rather than throw exceptions. The throwing exceptions behavior will return if enabling debug output. [TK-488](https://tickets.puppetlabs.com/browse/TK-488)
```
Comment (justinstoller) [inline documentation/release_notes.markdown:19]: That sounds terrible, who wrote it?   ( ;

File: documentation/release_notes.markdown [hunk 2]
```diff
@@ -10,6 +10,16 @@ canonical: "/puppetserver/latest/release_notes.html"
...
+
+- The metrics routes are now covered by trapperkeeper-authorization, so access can be controlled from auth.conf (or the authorization section in general in the tk configuration). There is still a default embedded jolokia-access.xml which restricts access to localhost by ip which needs to be overridden by software intending to use tk-auth instead. [TK-489](https://tickets.puppetlabs.com/browse/TK-489)
```
Comment (justinstoller) [inline documentation/release_notes.markdown:21]: people - users - intend to do thing, software not so much.

Comment (mwaggett) [reply documentation/release_notes.markdown:21]: what

File: documentation/release_notes.markdown (hunk 1 above)
Comment (Magisus) [reply documentation/release_notes.markdown:19]: "Jolokia will no longer log at debug level by default, which avoids large stacktraces for missing metrics. In order to re-enable debug output, set `debug` to `true` in `metrics.conf`, in addition to configuring the logging to `debug` in `logback.xml`."

File: documentation/release_notes.markdown (hunk 2 above)
Comment (Magisus) [reply documentation/release_notes.markdown:21]: "...by IP which needs to updated, in order to use tk-auth's rules instead."

Updated to allow anything?

File: documentation/release_notes.markdown (hunk 1 above)
Comment (mwaggett) [reply documentation/release_notes.markdown:19]: 💯

File: documentation/release_notes.markdown [hunk 3]
```diff
@@ -10,6 +10,16 @@ canonical: "/puppetserver/latest/release_notes.html"
...
 
+## Puppet Server 6.12.1
+
+Released 14 July 2020
+
+### Enhancements
+
+- Jolokia will no longer log at debug level by default, which avoids large stacktraces for missing metrics. In order to re-enable debug output, set `debug` to `true` in `metrics.conf`, in addition to configuring the logging to `debug` in `logback.xml`. [TK-488](https://tickets.puppetlabs.com/browse/TK-488)
+
+- The v2 metrics endpoint can now use trapperkeeper-authorization (tk-auth), which can be controlled from `auth.conf` (or from the authorization section of the trapperkeeper config). The v2 metrics endpoint is still restricted to localhost by default - if tk-auth is used to restrict access, you may override this behavior in `jolokia-access.xml`. [TK-489](https://tickets.puppetlabs.com/browse/TK-489)
```
Comment (mwaggett) [inline documentation/release_notes.markdown:21]: How does this sound?

File: documentation/release_notes.markdown [hunk 4]
```diff
@@ -10,6 +10,16 @@ canonical: "/puppetserver/latest/release_notes.html"
 [puppetserver.conf]: ./config_file_puppetserver.markdown
 [product.conf]: ./config_file_product.markdown
 
+## Puppet Server 6.12.1
+
+Released 14 July 2020
+
+### Enhancements
+
+- Jolokia will no longer log at debug level by default, which avoids large stacktraces for missing metrics. In order to re-enable debug output, set `debug` to `true` in `metrics.conf`, in addition to configuring the logging to `debug` in `logback.xml`. [TK-488](https://tickets.puppetlabs.com/browse/TK-488)
```
Comment (K8med) [inline documentation/release_notes.markdown:19]: Stack trace should, I believe, be two words and not one.

Comment (K8med) [inline documentation/release_notes.markdown:19]: The comma before "in addition to" can come out.

File: documentation/release_notes.markdown (hunk 3 above)
Comment (K8med) [inline documentation/release_notes.markdown:21]: The second line-- "The v2 metrics endpoint is still restricted to localhost by default - if tk-auth is used to restrict access, you may override this behavior in `jolokia-access.xml`"-- confused me a bit. Is the hyphen between "default" and "if" necessary? It is causing a funny read. Should it be 1) "The v2 metrics endpoint is still restricted to localhost by default if tk-auth is used to restrict access. You may override this behavior in `jolokia-access.xml`" or 2) "The v2 metrics endpoint is still restricted to localhost by default. If tk-auth is used to restrict access, you may override this behavior in `jolokia-access.xml`"

File: documentation/release_notes.markdown (hunk 4 above)
Comment (mwaggett) [reply documentation/release_notes.markdown:19]: ooh, yeah, I'm actually just gonna change it to "and configure the logging"

File: documentation/release_notes.markdown (hunk 3 above)
Comment (mwaggett) [reply documentation/release_notes.markdown:21]: should be #2, thanks for clarifying that! I use hyphens wildly and probably inappropriately haha. will update.

**Comment by mwaggett** [comment]
@K8med updated!

**Pull request merged**